### PR TITLE
Bug/jenkins 36851 hide favorites anonymous

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description

* See [JENKINS-36851](https://issues.jenkins-ci.org/browse/JENKINS-36851).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Apppropriate unit or acceptance tests or explaination to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

Log is as anonymous user and verify the following:
* Verify that dashboard cards do not display on main screen, and there is no favorite icon to the right of pipelines.
* Verify that Favorite star is not displayed in Pipeline Detail header.
* Verify that Favorite star is not displayed to the right of each branch on Pipeline Detail -> Branches tab.

- [x] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [x] Run the changes and verified the change matches the issue description
- [x] Reviewed the code
- [x] Verified that the appropriate tests have been written or valid explaination given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
@cliffmeyers this is ready for review?

**Comment by cliffmeyers** [comment]
Sorry I should flagged it as such. I was spending some time trying to write an automated test for this but some code coupling was making it more of a hassle than the value it ought to provide. :/

File: blueocean-personalization/src/main/js/redux/FavoritesStore.jsx [hunk 1]
```diff
@@ -30,7 +30,9 @@ function clone(json) {
 
 const actionHandlers = {
     [ACTION_TYPES.SET_USER](state, { payload }) {
-        const user = new User(payload);
+        const user = payload instanceof Error ?
```
Comment (michaelneale) [inline blueocean-personalization/src/main/js/redux/FavoritesStore.jsx]: so this has the side effect possibly that if there is a bug in fetching fav's, they may not just show up?

**Comment by michaelneale** [comment]
:bee: nice work!

**Pull request merged**